### ISSUE TITLE
notes and directions on how to improve the notebook

### ISSUE THREAD

**Issue by Krastanov** [adrianariton/qoptics#1]
I will use this issue to write up the directions for improvement that I have.

There will be multiple comments as I am reading through the note, please be careful to address all of them (it would be easy to miss some of them). Mark them as done as you go through them.

- [x] add a byline of what article this is about, who the authors of the article are, and who are the people that adapted it into a notebook.
- [x] add an introduction paragraph at the beginning explaining what this notebook will show
- [x] if you have found any typos in the original paper, double check you are not mistaken, and if you are right, just leave a note each time you have a correction

**Comment by Krastanov** [comment]
- [x] Add a table of content (either from PlutoUI or PlutoTeachingTools)

**Comment by Krastanov** [comment]
- [x] Every time you have a figure, explain why the figure is plotted and what the figure shows, e.g. write things like "Below we plot this expression with respect to ... We can see that ..."
- [x] There are some figures for which the span of the axes is very small and not particularly informative. For every figure, make sure that the span that you observe is showing something interesting.

**Comment by Krastanov** [comment]
- [x] this is a particularly important one: Make the titles of various subheadings tell a story that is easy for others to follow without knowing much about the paper (which is also why i am asking for a table of content). Maybe something like this (but please feel free to modify and improve):

> - Hamiltonian for a blue-detuned pump (because it is important that you are presenting the Hamiltonian, not that it is blue-detuned)
> - Analytical solution for time evolution and numerical comparison (with a few subheadings)
> - Rate of photon generation and of generating Bell pairs (and a subsection for visualizing Bell pair generation rate at various parameter settings)
> - Reviewing assumptions made and regimes in which they break (with a subsection for workarounds )

- [x] Also, say "source of infidelity" or "source of noise"  or "cause of ..." instead of "reason of infidelity". Also say "infidelity due to two-photon-excitations" instead of "2nd infidelity".

**Comment by Krastanov** [comment]
- [x] do not show the table output of the DiffEq.jl cell. The table in not informative or easy to read. Aim to visualize things if there is value to the reader in general -- that is the rule by which I am suggesting not printing out the table as the reader will not gain insight from the table
- [x] In the time evolution plots, do not plot real and imag separately, just plot the abs. Also could you elaborate on your choice of axes? I assume you simulated the time span that you have in order to be able to see the changes in both the blue and orange lines. If that is the case, I think you have made a pretty good tradeoff
- [x] Make the sliders and the graph modified by the sliders visible in the same window without scrolling 
- [x] make sure you do not have draft cells still visible (e.g. I am seeing `samples=23`)
- [x] Add a few "click the eye icon on the left in order to see or modify the code that is running" whenever there is hidden code. Maybe even use smaller font for it (there might be something in PlutoUI or PlutoTeachingTools that makes that more elegant)

**Comment by Krastanov** [comment]
- [x] In a few places you say "# of particles" -- that can be confusing for readers from the field -- please use "nb of photons in the pump mode" or something similar
- [x] instead of calculating the log10 of a value and then plotting it, use log-scaled axes - it renders a bit neater and is a bit easier to modify in the future
- [x] if you have rates of events, specify the unit (Hz maybe?)
- [x] I will repeat the guideline of explaining what the plots are in the surrounding text, because that is quite noticeable in the rate-vs-photons plots.
- [ ] there are a few things missing from the rate-vs-photons plots compared to the plot in the paper. Could you attempt to add a few of these missing lines? I will share a python notebook that might help with that shortly

**Comment by Krastanov** [comment]
Let's aim to have this finished by mid next week (maybe you can ping me with the updated version on Friday this week, I can do another quick review, and we can have it public by Wednesday next week). And after the second review on Friday we can start transitioning into more intricate simulations (for which I need to provide you with some more guidelines and code). Let me know if this seems like a reasonable time table?

I want to re-iterate that you are doing quite great job. It is very impressive to prepare a draft like this as quickly as you did.

**Comment by adrianariton** [comment]
Yes, it is perfect this way! We will be able to provide the updated version on Saturday morning, because on Friday we will need to move out luggages from Bucharest to our hometown, now that this college year is over, so I hope that's ok:)
Sorry for the inconvenience.

**Comment by Krastanov** [comment]
Great! Not an inconvenience at all! Whenever I suggest a timetable, do not hesitate to modify it if you have other constraints. The goal is for all of us to know when things are happening, not to chain you to an imposed schedule!

**Comment by adrianariton** [comment]
Ok perfect! Also could you please share that python notebook if possible, so we can get started on adding the missing purification graphs?

**Comment by adrianariton** [comment]
@Krastanov we've updated the notebook here on GitHub!

**Comment by Krastanov** [comment]
Thanks! A few more things to fix, some of them new issues:

- [x] "fidelity", not "accuracy" when talking about entanglement
- [x] similarly, "entangled Bell pairs", not "entangled particles" -- we do not really have particles in this setup
- [x] in English people usually say "initial condition", not "Cauchy problem condition"
- [x] instead of `md""""""` just put a `;` after the `end`
- [x] do not put the time interval as a slider that controls all plots -- varying that slider makes the notebook very slow (30 seconds to update). I would suggest fixing `tfin` to be 5 or 10 times the lifetime of the cavity `1/gamma_extrinsic`
- [x] something previously mentioned: do not say "particle number" or "particles". The readers would really have no idea that you are talking about "number of photons in the pump mode".
- [x] something previously mentioned: do not plot real and imaginary parts of `c` separately, just plot the abs -- make sure you do not have mismatch between the code, variable names, code comments, figure labels, and text. Currently there seems to be a bit of a mismatch in the first figure
- [x] in the first figure I see something about analytical vs numerical, but I see only the numerical line -- the analytical does not seem to be plotted
- [x] in the table above the first figure some of the numerical values are way too long: try to limit them to just a couple of digits after the decimal point. `Printf` might help here
- [x] plot the time in a more natural unit (e.g. a micro second or a millisecond), not .1us
- [x] is the MkHz variable used for anything? Clean up the ones that are not used if there are any. Define them closer to where they are used and comment them
- [ ] an in-place version of the ODE solver will probably be an order of magnitude faster. consider defining `f!` that uses `mul!` from `LinearAlgebra`

Let's try to have a more consistent set of numerical parameters and parameter ranges. For some of this I am suggesting steps that make precomputing every state of the sliders a bit cheaper and for some of these I am using background knowledge to make the parameter ranges of higher interest to the readers.

- [x] `g₀` should be between 10Hz and 100kHz in log-steps of factor of 1 (couplings of more than 10kHz are science fiction due to the difficulty of having a microwave photon and an optical photon interact (due to the difference in energy scales))
- [x] `nₚ` should be between 1 and 1000 in log-steps of 0.5 (it is basically impossible to put more than a 1000 photons in without everything melting)
- [x] `γₑ` should be between 1MHz and 10GHz in log-steps of 1 -- it is incredibly difficult, basically impossible, to make a cavity with leakage lower than 1MHz -- think of it this way: a photon frequency is 100THz, so 1MHz leakage is a Q factor of 10THz/1MHz, i.e. Q=10^8 -- that is incredibly difficult to make
- by log-step of 1, I mean a factor of 10 between steps; by log-step of 0.5, I mean a factor of sqrt(10) between steps
- [x] c_0 and c_1 are not very descriptive in plot legends. Use things like `|00⟩ population (analytical)` in the legend or something similarly descriptive
- [x] **this is pretty important:** Remove the `real` or `imag` branches for `gprime`. If gprime is not real, just raise a warning `@warn` and do not plot anything

I will continue with comments on Rate of photon generation and the rest in a separate comment

**Comment by adrianariton** [comment]
Hello @Krastanov !
I've finished the current tasks, hopefully I haven't missed another misexprimation.

Also some extra
## For the entanglement rate section
- added an option to center the graph properly, but also allowed the plot not to be cantered if the reader wants (for the plot to look similarly to the one in the paper proportion-wise). 
-  calculated the optimal pump power and number of photons in pump mode, and added them to the plot (under the `Guides` option). Also added a bit of text for that part in order to make the reader understand why we plot the optimal number of photons and also how we deduce it from the formulas (it is a pretty short comment before the plot, because the solution is pretty straight forward)
- Also redid some of the code following your advice (excluding the infidelity section which we will do tomorrow :) )
## Other stuff
- Modified all the sliders ,except the `∆t` one to which I will get to also tomorrow
- Also commented the important function in a cpp manner, let me know if that's ok
- Fixed most of the plotting issues (other than the infidelity plot)
- Also hopefully the updated inlace ODE is ok. Let me know if it isn't the case so I can redo it

**Comment by Krastanov** [comment]
Great! Could you also check Zulip? I sent a couple of messages there
